procurement_plan/models/procurement_plan.py

```diff
@@ -0,0 +1,186 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
+from openerp import models, fields, api, exceptions, _
+
+
+class ProcurementPlan(models.Model):
+
+    _name = 'procurement.plan'
+    _description = 'Procurement Plan'
+    _rec_name = 'sequence'
+
+    @api.one
+    @api.depends('procurement_ids', 'procurement_ids.state')
+    def _get_state(self):
+        self.state = 'running'
+        if not self.procurement_ids:
+            self.state = 'draft'
+        elif (len(self.procurement_ids.filtered(
+                lambda x: x.state == 'draft')) == len(self.procurement_ids)):
+            self.state = 'draft'
+        elif (len(self.procurement_ids.filtered(
```

Comment:
Esto se puede hacer mejor de esta forma:
`elif all(x.state == 'done' for x in self.procurement_ids):`